Build-FFmpeg.ps1

```diff
@@ -0,0 +1,205 @@
+
+    try {
+        MSBuild.exe $SolutionDir\Libs\bzip2\SMP\libbz2.vcxproj `
+            /p:OutDir="$libs\build\" `
+            /p:Configuration="${Configuration}WinRT" `
+            /p:Platform=$Platform `
+            /p:WindowsTargetPlatformVersion=$WindowsTargetPlatformVersion `
+            /p:PlatformToolset=$platformToolSet
+    }
+    catch {
+        Exit
+    }
+
+    Get-ChildItem -Recurse -Include '*.h' $libs\build\libbz2\include | Copy-Item -Destination $libs\include\
+    Copy-Item -Recurse $libs\build\libbz2\licenses\* -Destination $libs\licenses\
+    Copy-Item $libs\build\libbz2\lib\$Platform\libbz2_winrt.lib $libs\lib\bz2.lib
+    Copy-Item $libs\build\libbz2\lib\$Platform\libbz2_winrt.pdb $libs\lib\bz2.pdb
+
+    try {
+        MSBuild.exe $SolutionDir\Libs\libiconv\SMP\libiconv.vcxproj `
+            /p:OutDir="$libs\build\" `
+            /p:Configuration="${Configuration}WinRT" `
+            /p:Platform=$Platform `
+            /p:WindowsTargetPlatformVersion=$WindowsTargetPlatformVersion `
+            /p:PlatformToolset=$platformToolSet
+    }
+    catch {
+        Exit
+    }
+
+    Get-ChildItem -Recurse -Include '*.h' $libs\build\libiconv\include | Copy-Item -Destination $libs\include\
+    Copy-Item -Recurse $libs\build\libiconv\licenses\* -Destination $libs\licenses\
+    Copy-Item $libs\build\libiconv\lib\$Platform\libiconv_winrt.lib $libs\lib\iconv.lib
+    Copy-Item $libs\build\libiconv\lib\$Platform\libiconv_winrt.pdb $libs\lib\iconv.pdb
+
+    $env:LIB += ";${libs}\lib"
+    $env:INCLUDE += ";${libs}\include"
+
+    # Export full current PATH from environment into MSYS2
+    $env:MSYS2_PATH_TYPE = 'inherit'
+
+    # Build ffmpeg
+    & $Msys2Bin --login -x $SolutionDir\FFmpegConfig.sh Win10 $Platform
+
+    # Copy PDBs to built binaries dir
+    Get-ChildItem -Recurse -Include '*.pdb' $SolutionDir\ffmpeg\Output\Windows10\$Platform | `
+        Copy-Item -Destination $SolutionDir\ffmpeg\Build\Windows10\$Platform\bin\
+}
+
+if (! (Test-Path $PSScriptRoot\ffmpeg\configure)) {
+    Write-Error 'configure is not found in ffmpeg folder. Ensure this folder is populated with ffmpeg snapshot'
+    Exit
+}
+
+if (!(Test-Path $Msys2Bin)) {
+
+    $msysFound = $false
+    @( 'C:\msys64', 'C:\msys' ) | ForEach-Object {
+        if (Test-Path $_) {
+            $Msys2Bin = "${_}\usr\bin\bash.exe"
+            $msysFound = $true
+
+            break
+        }
+    }
+
+    # Search for MSYS locations
+    if (! $msysFound) {
+        Write-Error "MSYS2 not found."
+        Exit;
+    }
+}
+
+[System.IO.DirectoryInfo] $vsLatestPath = `
+    & "$VSInstallerFolder\vswhere.exe" `
+    $VsWhereCriteria `
+    -property installationPath `
+    -products * `
+    -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
+
+Write-Host "Visual Studio Installation folder: [$vsLatestPath]"
+
+$start = Get-Date
+
+foreach ($platform in $Platforms) {
+    Build-Platform `

```

Comment:
Will the script be terminated if one of the platform builds fails?

Reply:
Yes.